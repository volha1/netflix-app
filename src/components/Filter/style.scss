@import '../../common/scss/base.scss';

.filter {
  display: flex;
  justify-content: space-between;
  padding: 20px 0px;
  border-bottom: 2px solid $gray;
}
