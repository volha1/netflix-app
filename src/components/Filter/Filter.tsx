import React, { ReactElement, Dispatch, SetStateAction } from 'react';
import GenreFilter from '../GenreFilter';
import Sorting from '../Sorting';
import { genres } from '../../helpers/constants';
import './style.scss';
import SearchParams from '../../types/SearchParams';

const genreValues = ['All', ...genres];

type FilterProps = {
  params: SearchParams;
  setParams: Dispatch<SetStateAction<SearchParams>>;
  removeSearchParams: Dispatch<SetStateAction<string>>;
};

const Filter = ({ params, setParams, removeSearchParams }: FilterProps): ReactElement => {
  return (
    <div className="filter content">
      <GenreFilter params={params} onFilter={setParams} genres={genreValues} removeSearchParams={removeSearchParams} />
      <Sorting params={params} onSort={setParams} />
    </div>
  );
};

export default Filter;
