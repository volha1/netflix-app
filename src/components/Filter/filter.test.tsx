import React from 'react';
import renderer from 'react-test-renderer';

import Filter from './Filter';

describe('Filter test', () => {
  const setParams = jest.fn();
  const removeSearchParams = jest.fn();
  const params = { filter: '', sortOrder: '', sortBy: '', search: '', movie: '' };

  it('Filter snapshot', () => {
    const component = renderer.create(
      <Filter setParams={setParams} removeSearchParams={removeSearchParams} params={params} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
