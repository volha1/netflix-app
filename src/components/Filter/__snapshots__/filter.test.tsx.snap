// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter test Filter snapshot 1`] = `
<div
  className="filter content"
>
  <ul
    className="genres-list"
  >
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      documentary
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      comedy
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      horror
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      crime
    </button>
  </ul>
  <div
    className="sorting"
  >
    <label
      className="sorting-label"
      htmlFor="params"
    >
      Sort by
    </label>
    <select
      defaultValue="default"
      id="params"
      name="params"
      onChange={[Function]}
      value="_"
    >
      <option
        disabled={true}
        value="default"
      >
        Choose here...
      </option>
      <option
        value="release_date_asc"
      >
        Release date (asc)
      </option>
      <option
        value="release_date_desc"
      >
        Release date (desc)
      </option>
      <option
        value="vote_average_asc"
      >
        Rating (asc)
      </option>
      <option
        value="vote_average_desc"
      >
        Rating (desc)
      </option>
    </select>
  </div>
</div>
`;
